app/assets/javascripts/components/forms/validations.jsx

```diff
@@ -13,20 +14,48 @@ export function signalNameValidator(name) {
 }
 
 export function defaultResponseValidator(defaultResponse) {
-  if (!defaultResponse) {
-    return 'Default response required.';
-  } else if (defaultResponse.length > 140) {
+  return responseTextValidator(defaultResponse, 'Default');
+}
+
+export function repeatResponseValidator(repeatResponse) {
+  return responseTextValidator(repeatResponse, 'Repeat');
+}
+
+export function timedResponseValidator(responses) {
+  const responseArrayErrors = [];
+
+  if (responses.length) {
+    responses.forEach((response, index) => {
+      const responseErrors = {};
+      responseErrors.message = responseTextValidator(response.message, 'Timed');
+      responseErrors.expiration_date = expirationDateValidator(response.expiration_date);
+      responseArrayErrors[index] = responseErrors;
+    });
+  }
+
+  return responseArrayErrors.length ? responseArrayErrors : null;
+}
+
+export function responseTextValidator(text, type) {
+  if (!text) {
+    return `${type} response reqired.`;
+  }
+
+  if (text.length > 140) {
     return 'Your tweet response cannot exceed 140 characters.';
   }
 
   return null;
 }
 
-export function repeatResponseValidator(repeatResponse) {
-  if (!repeatResponse) {
-    return 'Repeat respose reqired.';
-  } else if (repeatResponse.length > 140) {
-    return 'Your tweet response cannot exceed 140 characters.';
+export function expirationDateValidator(date) {
+  if (!date) {
+    return 'Expiration date must be set on a custom response';
+  }
+
+  const selectedDate = moment(date, 'YYYY-MM-DD');
+  if (selectedDate < moment()) {
+    return 'Expiration date cannot be in the past';
```

Comment:
What about custom responses that are supposed to be expired? Do you only want to do this for new ones?